### PR TITLE
Add sasl username to abuser report

### PR DESCRIPTION
**Pull request by the-eater**
this PR adds
- option to select an selector for the abuse report, the current options are `sasl_username` and `sender_domain`
- the column `Sasl Username` to the abusers report
- A key to the `session` for the column `sasl_username`, for the grouping in the abusers report

### CODE REVIEW

File: src/cluegetter/http.go [hunk 1]
```diff
@@ -412,14 +419,28 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
...
+	data.Selectors = append(data.Selectors, httpAbuserSelector{
+		Name:     "sasl_username",
+		Text:     "Sasl Username",
+		Selected: r.FormValue("selector") == "sasl_username",
+	})
+
+	data.Selectors = append(data.Selectors, httpAbuserSelector{
+		Name:     "sender_domain",
+		Text:     "Sender domain",
+		Selected: r.FormValue("selector") != "sasl_username",
```
Comment (Freeaqingme) [inline src/cluegetter/http.go]: == "sender_domain"

File: src/cluegetter/http.go [hunk 2]
```diff
@@ -412,14 +419,28 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
...
+	})
+
```
Comment (Freeaqingme) [inline src/cluegetter/http.go:439]:
```
    http.Error(w, err.Error(), http.StatusBadRequest)
```

File: src/cluegetter/http.go [hunk 3]
```diff
@@ -412,14 +419,28 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
...
 	}{
 		httpGetViewData(),
 		httpGetInstances(),
 		"4",
 		"5",
 		make([]*httpAbuserTop, 0),
+		make([]httpAbuserSelector, 0),
 	}
 
+	data.Selectors = append(data.Selectors, httpAbuserSelector{
```
Comment (Freeaqingme) [inline src/cluegetter/http.go]: Extract to own method.

**Comment by Freeaqingme** [comment]
@EaterOfCode when do you think you can implement the feedback?

**Comment by the-eater** [comment]
@Freeaqingme all points should be fixed now, feedback is welcome

File: src/cluegetter/http.go [hunk 4]
```diff
@@ -458,13 +480,43 @@ func httpAbusersHandler(w http.ResponseWriter, r *http.Request) {
...
+	hasSelectedSelector := false
+	for _, selector := range selectors {
+		if selector.Selected {
+			hasSelectedSelector = true
+			break
+		}
+	}
+
+	if !hasSelectedSelector && r.FormValue("selector") != "" {
+		return nil, errors.New("'" + r.FormValue("selector") + "' is not a valid value for parameter selector")
```
Comment (Freeaqingme) [inline src/cluegetter/http.go]: Sorry for not noticing earlier. The conventino is to first display what's wrong, then show the specified value. E.g.:

Invalid selector specified: &lt;selector>

Now you also get to add a capital at the start of the sentence #winrar :D

**Comment by Freeaqingme** [comment]
Besides that 1 last comment it's good to go.

**Comment by the-eater** [comment]
@Freeaqingme last comment is fixed :)

**Pull request merged**